Source/Particles/Collision/BinaryCollision/Coulomb/ElasticCollisionPerez.H

```diff
@@ -108,13 +115,21 @@ void ElasticCollisionPerez (
           u1y[I1[i1]] = u1xbuf*std::sin(theta) + u1y[I1[i1]]*std::cos(theta);
 #endif
 
+          // compute the effective density used for s12 in UpdateMomentumPerezElastic()
+          // The effective density used for s12 is defined such that the average value of s12
+          // after many scattering cycles is the same as that for the full NxN pairing method.
+          // The approach here is similar to that in JCP 413 (2020) by Higginson et al., but
+          // does not use a unique duplication factor for each macro-particle pair.
+          T_PR n12;
+          const T_PR wpmax = amrex::max(w1[ I1[i1] ],w2[ I2[i2] ]);
+          if (isSameSpecies) { n12 = wpmax*static_cast<T_PR>(min_N+max_N-1)/dV; }
+          else { n12 = wpmax*static_cast<T_PR>(min_N)/dV; }
```

Comment:
The factor `min_N` is not explained in Higginson et al. (since Higginson's paper is duplicating particles instead, which results in a different factor). 

My understanding is that, for a given species (say species 1), this factor should correspond to the number of macroparticles in the other species (`N2` for species 1), in order to account for the fact that a macroparticle from species 1 will only collide with one or (a few) macroparticles of species 2, instead of all of the `N2` macroparticles. 
If species 1 is the species with the larger number of macroparticles, this automatically works (`min_N = N2`).
If species 2 is the species with the lowest number of macroparticles, then its particles will participate, on average, in `max_N/min_N` pairs, so it is as if they see a factor `max_N/min_N x min_N = max_N = N2`, so it works in this case.

Is the above correct? If yes, could you add an additional comment explaining the factor `min_N` in the code (if possible, in a more concise and clearer way than the explanation I gave above).

Reply:
Note also that the other binary collisions in WarpX use a duplication method similar to Higginsons' paper:

https://github.com/ECP-WarpX/WarpX/blob/development/Source/Particles/Collision/BinaryCollision/DSMC/DSMCFunc.H#L147
https://github.com/ECP-WarpX/WarpX/blob/development/Source/Particles/Collision/BinaryCollision/DSMC/DSMCFunc.H#L173

https://github.com/ECP-WarpX/WarpX/blob/development/Source/Particles/Collision/BinaryCollision/NuclearFusion/NuclearFusionFunc.H#L186
https://github.com/ECP-WarpX/WarpX/blob/development/Source/Particles/Collision/BinaryCollision/NuclearFusion/NuclearFusionFunc.H#L173

Thus, it might be more consistent to use Higginson's duplication method in the Coulomb module too.

Are there advantages to the proposed method without duplication, aside from the slightly simpler implementation?

Reply:
It is not just Nmax vs Nmin that is the difference. In the Higginson method, one uses (wmax_prime)*Nmax, where wmax_prime = max(wi/df,wj/df) is the max modified weight of the two particles with dp the duplication factor. Note that the average duplication factor per pair is Nmax/Nmin, which means on average (wmax_prime) *Nmax = wmax *Nmin.

Using Nmin is certainly simpler because you don't have to used a modified weight that takes into account a pair-specific duplication factor. It is also the most direct extension of the original TA77 method, which uses Nmin.

Reply:
> The factor `min_N` is not explained in Higginson et al. (since Higginson's paper is duplicating particles instead, which results in a factor `max_N` instead).
> 
> My understanding is that, for a given species (say species 1), this factor should correspond to the number of macroparticles in the other species (`N2` for species 1), in order to account for the fact that a macroparticle from species 1 will only collide with one or (a few) macroparticles of species 2, instead of all of the `N2` macroparticles. If species 1 is the species with the larger number of macroparticles, this automatically works (`min_N = N2`). If species 2 is the species with the lowest number of macroparticles, then its particles will participate, on average, in `max_N/min_N` pairs, so it is as if they see a factor `max_N/min_N x min_N = max_N = N2`, so it works in this case.
> 
> Is the above correct? If yes, could you add an additional comment explaining the factor `min_N` in the code (if possible, in a more concise and clearer way than the explanation I gave above).

Your interpretation is correct. It is also exactly the same interpretation of the original method for equally-weighted particles by Takizuka and Abe 1977.

Reply:
You asked about advantages of the proposed method.

As you stated, one advantage is that it is simpler. Rather than adding un-necessary lines of code here to maintain quasi-consistency with DSMC and Fusion methods, I would suggest considering adopting the simplified approach here in those modules. In addition to being simpler, there is in fact another advantage when using it for reactions that result in product species such as Fusion. I’ll try to concisely explain an issue with the current approach used in the fusion module below.

For simplicity, consider inter-species fusion between two species with equal weight particles w and using fusion_multiplier = 1. Using N_max, the reaction probability will be too large by N_max/N_min on average. However, because the reaction weight used is w_min = N_min/N_max*w (on average), the total number of fusion products produced is correct on average. So, the current method essentially uses a fusion_multiplier of N_max/N_min if even the specified fusion_multiplier is 1. Thus, it is not possible to do a simulation with equally weighted particles as the particles will no longer be equally weighted after a single fusion event when N1 and N2 are not equal. This approach can also result in the avalanching production of small weight particles described in PR #4976 

The alternative approach has an advantage in that it allows one to maintain equally-weighted particles and it will not have the issue described in PR #4976.

I'm happy to discuss further off line if needed.

Reply:
I agree we should switch the DSMC and fusion modules to also use this new method. Doing that for DSMC is on my to-do list.